core/src/plugins/filed/python/percona-xtrabackup/bareos-fd-percona-xtrabackup.py

```diff
@@ -394,27 +400,57 @@ def plugin_io(self, IOP):
             return bRC_OK
 
         elif IOP.func == IO_CLOSE:
-            DebugMessage(100, "plugin_io called with IO_CLOSE\n")
-            self.subprocess_returnCode = self.stream.poll()
-            if self.subprocess_returnCode is None:
-                # Subprocess is open, we wait until it finishes and get results
-                try:
-                    self.stream.communicate()
-                    self.subprocess_returnCode = self.stream.poll()
-                except:
-                    JobMessage(
-                        M_ERROR,
-                        "Dump / restore command not finished properly\n",
-                    )
-                    bRC_Error
-                return bRC_OK
-            else:
+            DebugMessage(100, "plugin_io: called with IO_CLOSE\n")
+
+            if self.jobType == "B":
                 DebugMessage(
                     100,
-                    "Subprocess has terminated with returncode: %d\n"
-                    % self.subprocess_returnCode,
+                    "plugin_io: calling end_dumper() to wait for PID %s to terminate\n"
+                    % (self.stream.pid),
                 )
-                return bRC_OK
+                bareos_xtrabackup_dumper_returncode = self.end_dumper()
+                if bareos_xtrabackup_dumper_returncode != 0:
+                    JobMessage(
+                        M_FATAL,
+                        (
+                            "plugin_io[IO_CLOSE]: bareos_xtrabackup_dumper returncode:"
+                            " %s\n"
+                        )
+                        % (bareos_xtrabackup_dumper_returncode),
+                    )
+
+                    # Cleanup tmpdir
+                    shutil.rmtree(self.tempdir)
+                    # Set return code to 9 because we've terminated subprocess earlier
```

Comment:
```suggestion
```
Seems to be obsolete here, should be done in `end_dumper()`, see suggestion below.